cmd/soroban-rpc/internal/db/db.go

```diff
@@ -23,35 +22,24 @@ var migrations embed.FS
 var ErrEmptyDB = errors.New("DB is empty")
 
 const (
-	ledgerEntriesTableName      = "ledger_entries"
 	metaTableName               = "metadata"
 	latestLedgerSequenceMetaKey = "LatestLedgerSequence"
 )
 
-type DB interface {
-	NewLedgerEntryUpdaterTx(forLedgerSequence uint32, maxBatchSize int) (LedgerEntryUpdaterTx, error)
-	NewLedgerEntryReaderTx() (LedgerEntryReaderTx, error)
-	io.Closer
-}
-
-type LedgerEntryUpdaterTx interface {
-	UpsertLedgerEntry(key xdr.LedgerKey, entry xdr.LedgerEntry) error
-	DeleteLedgerEntry(key xdr.LedgerKey) error
-	Done() error
-}
-
-type LedgerEntryReaderTx interface {
-	GetLatestLedgerSequence() (uint32, error)
-	GetLedgerEntry(key xdr.LedgerKey) (bool, xdr.LedgerEntry, error)
-	Done() error
+type Writer interface {
+	NewTx(ctx context.Context, maxBatchSize int) (WriteTx, error)
+	GetLatestLedgerSequence(ctx context.Context) (uint32, error)
+	WALCheckpoint(ctx context.Context) error
```

Comment:
It would be great if could keep this SQL-agnostic (and this is sqlite specific).